tests/PHPStan/Rules/PhpDoc/data/bug-12457.php

```diff
@@ -0,0 +1,15 @@
+<?php declare(strict_types = 1);
+
+namespace Bug12457;
+
+class HelloWorld
+{
+	/**
+	 * @param array{numeric-string&uppercase-string&lowercase-string} $a
+	 */
+	public function sayHello(array $a): void
+	{
+		/** @var array{numeric-string} $b */
+		$b = $a;
+	}
```

Comment:
do we need more tests for different variants of this problem?

like

```suggestion
	}
	
	/**
	 * @param array{numeric-string&uppercase-string&lowercase-string} $a
	 */
	public function sayUpper(array $a): void
	{
		/** @var array{uppercase-string} $b */
		$b = $a;
	}

	/**
	 * @param array{numeric-string&uppercase-string&lowercase-string} $a
	 */
	public function sayLower(array $a): void
	{
		/** @var array{lowercase-string} $b */
		$b = $a;
	}
```

Reply:
maybe also variants with `non-empty-string`, `non-falsey-string`?

Reply:
I don't think so since it's not related to numeric-string directly.

The issue was because it was a ConstantArray with a lowercase-string or upercase-string inside.
Because of the early `return $type` we didn't process the content of the array.

Reply:
But I added a test for the `callable` case
https://github.com/phpstan/phpstan-src/pull/3792/commits/ad97094494cfa5ecb6191c3670f5276a4b8e1770